x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/gen/script/Scripts.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.expression.gen.script;
+
+import org.elasticsearch.xpack.sql.expression.function.scalar.whitelist.InternalSqlScriptUtils;
+import org.elasticsearch.xpack.sql.type.DataType;
+
+import java.util.LinkedHashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+import static org.elasticsearch.xpack.sql.expression.gen.script.ParamsBuilder.paramsBuilder;
+
+public final class Scripts {
+
+    private Scripts() {}
+
+    private static final Map<Pattern, String> FORMATTING_PATTERNS;
+
+    static {
+        Map<String, String> patterns = new LinkedHashMap<>();
```

Comment:
Suggestion: You could do it in one step without the intermediate `patterns` map.

Maybe you can use the `ImmutableMap.of()` or
```
Collections.unmodifiableMap(Stream.of(
    new SimpleEntry<>(Pattens.compile("doc[{}].value", Pattern.LITERAL), "{sql}.docValue(doc,{})"),
    ...
    .collect(Collectors.toMap((e) -> e.getKey(), (e) -> e.getValue())));
``` and avoid the `static` block.

Reply:
We're trying to stay away from Guava especially for simple things.
The `Stream` suggestion is good however after adding it I think it decreases readability (there's a lot of `Pattern.compile` repetition and I find the map collector complicated).
The `static` block might be a bit more verbose but I find it clearer.

Reply:
No worries, was just a suggestion.